ste/mgr-JobMgr.go

```diff
@@ -571,13 +571,25 @@ func (jm *jobMgr) reportJobPartDoneHandler() {
 
 	for {
 		select {
-		case <-jm.reportCancelCh:
+		case _, ok := <-jm.reportCancelCh:
+			if ok {
+				jm.Panic(fmt.Errorf("Received message on reportCancelCh instead of close signal"))
```

Comment:
add a comment above this

// We close jm.reportCancelCh to convey termination of jobMgr (see cleanupTransferRoutine()) so we MUST never read anything from jm.reportCancelCh

and the following comment above L583
// We strictly MUST terminate jobMgr after job completes, so we MUST never get a close signal on jm.reportCancelCh while job is not completed.